src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -341,6 +341,11 @@ private void CreateNodeCaches()
 
         public IEETypeNode NecessaryTypeSymbol(TypeDesc type)
         {
+            if (!_compilationModuleGroup.IsSingleFileCompilation && EETypeNode.IsTypeNodeShareable(type))
```

Comment:
You can skip attempts to generate constructed generic type definition EETypes by calling `ConstructedEETypeNode.CreationAllowed` (it will also skip attempts to make them for unmanaged and managed pointers, which is good to prevent asserts too). We use `CreationAllowed` in all the places where we would like to generate the "EEType with the highest possible load level".